app/Actions/Album/Archive.php

```diff
@@ -99,7 +101,7 @@ private static function createZipTitle(Collection $albums): string
 	 */
 	private static function createValidTitle(string $title): string
 	{
-		return str_replace(self::BAD_CHARS, '', $title) ?? 'Untitled';
+		return str_replace(self::BAD_CHARS, '', $title) ?: 'Untitled';
```

Comment:
So what's up with this? `phpstan-strict-rules` explicitly bans `?:` although it looks like you filtered that test out? Still, why replace the existing instance of a null coalescing operator that `phpstan-strict-rules` recommends?

Reply:
Because ?? uses comparison against null.
In this specific case you want a comparison against false.

```php
$x = "";
echo ($x ?: "hello"); # "hello"
echo ($x ?? "hello"); # ""
```

Reply:
I wonder if this discussion is not a prime example against using `empty` and short ternary syntax and why PhpStan forbids them by default. With all PhpStan rules enabled we had to rewrite it to

```suggestion
		$sanitizedTitle =  str_replace(self::BAD_CHARS, '', $title);
	
		return $sanitizedTitle !== false && $sanitizedTitle !== '' &&  ? sanitizedTitle : 'Untitled';
```

Yes, it is very verbose, maybe even too verbose for my taste, but it makes _very_ explicit what happens.

Reply:
I think you meant:
```suggestion
		$sanitizedTitle =  str_replace(self::BAD_CHARS, '', $title);
		return $sanitizedTitle !== '' ? sanitizedTitle : 'Untitled';
```

Reply:
Ahm, yes.